tests/tests/metrics/t10_server_metrics_and_data_metrics.rs

```diff
@@ -64,6 +66,111 @@ async fn server_metrics_and_data_metrics() -> Result<()> {
     Ok(())
 }
 
+/// Test if heartbeat metrics work
+#[async_entry::test(worker_threads = 8, init = "init_default_ut_tracing()", tracing_span = "debug")]
+async fn heartbeat_metrics() -> Result<()> {
+    // Setup test dependencies.
+    let config = Arc::new(
+        Config {
+            enable_heartbeat: false,
+            heartbeat_interval: 50,
+            enable_elect: false,
+            ..Default::default()
+        }
+        .validate()?,
+    );
+    let mut router = RaftRouter::new(config.clone());
+
+    tracing::info!("--- initializing cluster");
+    let log_index = router.new_cluster(btreeset! {0,1,2}, btreeset! {}).await?;
+
+    tracing::info!(
+        log_index,
+        "--- heartbeat disabled, interval since the last ack should increase in flush loop"
+    );
+    let leader = router.get_raft_handle(&0)?;
+    {
+        let mut data_metrics = leader.data_metrics();
+        let mut prev_node1 = 0;
+        let mut prev_node2 = 0;
+        for _ in 0..30 {
```

Comment:
Here, since the heartbeat is disabled, the last ack time of node 1 and node 2 should never change, so I expect the time since the last ack should increase during the loop, but it won't sometimes, I am not sure about the reason:<

```
running 1 test
panicked at tests/tests/metrics/t10_server_metrics_and_data_metrics.rs:109:13:
interval should increase since last ack won't change
tests/tests/metrics/t10_server_metrics_and_data_metrics.rs:109:13
backtrace is disabled without --features 'bt'

test t10_server_metrics_and_data_metrics::heartbeat_metrics ... FAILED
```

Reply:
Any change to `RaftMetrics` would result in `changed()` to be ready, it is possible other fields changed but the `heartbeat` did not change.

I dumped the node-0 metrics from the [log](https://github.com/datafuselabs/openraft/actions/runs/10001717998?pr=1177) (at the bottom of the page):
`cat ut.2024-07-19-02 | grep heartbeat_metrics | grep report_metrics:.* -o  | grep id:0`

But it looks lite `impl Display for RaftMetrics` is not updated I can not see the `heartbeat` field:
```
report_metrics: Metrics{id:0, Learner, term:0, vote:<T0-N0:->, last_log:None, last_applied:None, leader:None(since_last_ack:None ms), membership:{log_id:None, {voters:[], learners:[]}}, snapshot:None, purged:None, replication:{}}
report_metrics: Metrics{id:0, Learner, term:0, vote:<T0-N0:->, last_log:None, last_applied:None, leader:None(since_last_ack:None ms), membership:{log_id:None, {voters:[], learners:[]}}, snapshot:None, purged:None, replication:{}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:1, last_applied:None, leader:0(since_last_ack:0 ms), membership:{log_id:T0-N0.0, {voters:[{0:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.1}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:1, last_applied:T1-N0.1, leader:0(since_last_ack:0 ms), membership:{log_id:T0-N0.0, {voters:[{0:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.1}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:2, last_applied:T1-N0.1, leader:0(since_last_ack:0 ms), membership:{log_id:T1-N0.2, {voters:[{0:()}], learners:[1:()]}}, snapshot:None, purged:None, replication:{0:T1-N0.2,1:None}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:2, last_applied:T1-N0.2, leader:0(since_last_ack:0 ms), membership:{log_id:T1-N0.2, {voters:[{0:()}], learners:[1:()]}}, snapshot:None, purged:None, replication:{0:T1-N0.2,1:None}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:3, last_applied:T1-N0.2, leader:0(since_last_ack:0 ms), membership:{log_id:T1-N0.3, {voters:[{0:()}], learners:[1:(),2:()]}}, snapshot:None, purged:None, replication:{0:T1-N0.3,1:None,2:None}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:3, last_applied:T1-N0.3, leader:0(since_last_ack:0 ms), membership:{log_id:T1-N0.3, {voters:[{0:()}], learners:[1:(),2:()]}}, snapshot:None, purged:None, replication:{0:T1-N0.3,1:None,2:None}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:3, last_applied:T1-N0.3, leader:0(since_last_ack:0 ms), membership:{log_id:T1-N0.3, {voters:[{0:()}], learners:[1:(),2:()]}}, snapshot:None, purged:None, replication:{0:T1-N0.3,1:T1-N0.3,2:T1-N0.3}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:4, last_applied:T1-N0.3, leader:0(since_last_ack:14 ms), membership:{log_id:T1-N0.4, {voters:[{0:()},{0:(),1:(),2:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.4,1:T1-N0.3,2:T1-N0.3}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:4, last_applied:T1-N0.4, leader:0(since_last_ack:10 ms), membership:{log_id:T1-N0.4, {voters:[{0:()},{0:(),1:(),2:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.4,1:T1-N0.3,2:T1-N0.4}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:5, last_applied:T1-N0.4, leader:0(since_last_ack:19 ms), membership:{log_id:T1-N0.5, {voters:[{0:(),1:(),2:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.5,1:T1-N0.4,2:T1-N0.4}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:5, last_applied:T1-N0.5, leader:0(since_last_ack:9 ms), membership:{log_id:T1-N0.5, {voters:[{0:(),1:(),2:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.5,1:T1-N0.4,2:T1-N0.5}}
report_metrics: Metrics{id:0, Leader, term:1, vote:<T1-N0:Q>, last_log:5, last_applied:T1-N0.5, leader:0(since_last_ack:8 ms), membership:{log_id:T1-N0.5, {voters:[{0:(),1:(),2:()}], learners:[]}}, snapshot:None, purged:None, replication:{0:T1-N0.5,1:T1-N0.5,2:T1-N0.5}}
```

Reply:
> Any change to RaftMetrics would result in changed() to be ready, it is possible other fields changed but the heartbeat did not change.

Actually and surprisingly, the interval value in the next loop can even be smaller than the previous one:

```sh
$ cargo t t10_server_metrics_and_data_metrics::heartbeat_metrics -- --nocapture
running 1 test
heartbeat: {0: Some(2), 1: Some(9), 2: Some(6)}
heartbeat: {0: Some(2), 1: Some(6), 2: Some(5)}
panicked at tests/tests/metrics/t10_server_metrics_and_data_metrics.rs:111:13:
interval should increase since last ack won't change
tests/tests/metrics/t10_server_metrics_and_data_metrics.rs:111:13
backtrace is disabled without --features 'bt'

test t10_server_metrics_and_data_metrics::heartbeat_metrics ... FAILED
```

> But it looks lite impl Display for RaftMetrics is not updated I can not see the heartbeat field:

Sorry for missing that, it has just been added:)

Reply:
Hmmm, if another AppendEntries or heartbeat response is received, this interval will become smaller.

Reply:
Heartbeat is disabled in this test cluster, so there shouldn't be heartbeat response. Then I guess it is made by a replication during the cluster initialization?

Reply:
I removed this test, do we more tests or the existing 2 tests are sufficient, let me know what you think:)